opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpanSource.java

```diff
@@ -36,7 +34,12 @@ public ThreadLocalActiveSpan activeSpan() {
 
     @Override
     public ActiveSpan makeActive(Span span) {
-        return new ThreadLocalActiveSpan(this, span, new AtomicInteger(1));
+        return new ThreadLocalActiveSpan(this, span, new AutoFinisher());
```

Comment:
perhaps this behavior should be controlled by a configuration parameter in ThreadLocalActiveSpanSource. Otherwise it seems to defeat the purpose of this PR, which was to give people a way to turn off refcounting / auto-finishing behavior.

Another thing that's unclear to me - those people who asked for no refcounting / auto-finish behavior, are they implementing a Tracer or doing instrumentation of some framework? Because if it's the latter and those frameworks are mixed with other instrumentation that expects auto-finish, then the only way to make this work is to have `makeActive(span) == makeActive(span, NoopObserver)`, i.e. if someone wants auto-finish behavior it must be requested explicitly, not via the Source configuration (it will be a breaking change from 0.30)